ios_widget_catalog_compare/lib/main.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/cupertino.dart';
+
+const EventChannel _platformEventChannel =
+    const EventChannel('overlay_ios.flutter.io/responder');
+
+void main() {
+  runApp(
+    CupertinoApp(
+      title: 'Flutter Demo',
+      home: FlutterDemo(),
+    ),
+  );
+}
+
+class FlutterDemo extends StatefulWidget {
+  FlutterDemo({Key key}) : super(key: key);
+
+  @override
+  _FlutterDemoState createState() => _FlutterDemoState();
+}
+
+class _FlutterDemoState extends State<FlutterDemo> {
+  String controlName = 'Null';
+
+  @override
+  void initState() {
+    super.initState();
+
+    _platformEventChannel.receiveBroadcastStream().listen((dynamic name) {
+      if (name != controlName) setState(() => controlName = name);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      child: Center(
+        child: widgetForKey(controlName),
+      ),
+    );
+  }
+
+  Widget widgetForKey(String key) {
+    Map<String, Widget> map = {
```

Comment:
you don't have to change this since it doesn't really affect anything, but micro-optimization for the future: you're actually building 3 things here and throwing 2 away. Again, doesn't really matter since widget declarations are very lightweight.

Reply:
Do you suggest I do a switch instead?

Reply:
moved the map up